src/core/environment.go

```diff
@@ -533,9 +533,7 @@ func (env *EnvironmentType) Processes() (result []*Process) {
 		name, _ := p.NameWithContext(ctx)
 		cmd, _ := p.CmdlineWithContext(ctx)
 
-		if name == "nginx" && !strings.Contains(cmd, "upgrade") {
-			nginxProcesses[pid] = p
-		}
+		nginxProcesses = env.getNginxProcess(nginxProcesses, name, pid, p, cmd)
```

Comment:
Could you just pass nginxProcesses & p to the function as arguments?
You could move the 3 lines of code above to the getNginxProcess function.